communication/nats/dialog/dialog_waiter.go

```diff
@@ -119,8 +120,20 @@ func (waiter *dialogWaiter) ServeDialogs(dialogHandler communication.DialogHandl
 	}
 	codec := NewCodecSecured(communication.NewCodecJSON(), waiter.signer, identity.NewVerifierSigned())
 	receiver := nats.NewReceiver(waiter.address.GetConnection(), codec, waiter.address.GetTopic())
+	return receiver.Respond(&dialogCreateConsumer{Callback:createDialog})
+}
 
-	return receiver.Respond(&dialogCreateConsumer{createDialog})
+// handleDialogStatus listens to peer dialog communication status and stores it local storage.
+func (waiter *dialogWaiter) handleDialogStatus(dialog communication.Dialog, peerID string) {
+	cb := func(r *communication.DialogStatusRequest) error {
+		msg := fmt.Sprintf("Peer '%s' dialog status code: %d, message: %s", peerID, r.Code, r.Message)
+		log.Info(waiterLogPrefix, msg)
+		// TODO: Log dialog status to local storage so it can be displayed in UI later.
```

Comment:
Not sure if we need it. Most likely we would need to store some last xxx sessions, just to troubleshoot things if needed. That can be saved in-memory only. Our policy is no-logs policy, such additional storage complicates issues with state storage and is also against our no-logs policy.